### PR TITLE
perf: don't re-create meter and instruments on every calls

### PR DESCRIPTION
**Pull request by bnjjj**
The creation of otel instruments using a regex is no longer part of the hot path. Now we create these instruments when starting the telemetry plugin and not in every serives.

---

**Checklist**

Complete the checklist (and note appropriate exceptions) before the PR is marked ready-for-review.

- [x] Changes are compatible[^1]
- [ ] Documentation[^2] completed
- [x] Performance impact assessed and acceptable
- Tests added and passing[^3]
    - [x] Unit Tests
    - [x] Integration Tests
    - [x] Manual Tests

**Exceptions**

*Note any exceptions here*

**Notes**

[^1]: It may be appropriate to bring upcoming changes to the attention of other (impacted) groups. Please endeavour to do this before seeking PR approval. The mechanism for doing this will vary considerably, so use your judgement as to how and when to do this.
[^2]: Configuration is an important part of many changes. Where applicable please try to document configuration examples.
[^3]: Tick whichever testing boxes are applicable. If you are adding Manual Tests, please document the manual testing (extensively) in the Exceptions.

### CODE REVIEW

File: apollo-router/src/plugins/telemetry/config_new/instruments.rs [hunk 1]
```diff
@@ -100,8 +116,118 @@ impl InstrumentsConfig {
             .defaults_for_levels(self.default_requirement_level, TelemetryDataKind::Metrics);
     }
 
-    pub(crate) fn new_router_instruments(&self) -> RouterInstruments {
+    pub(crate) fn new_static_router_instruments(&self) -> HashMap<String, StaticInstrument> {
```
Comment (BrynCooke) [inline apollo-router/src/plugins/telemetry/config_new/instruments.rs:119]: Can we consider changing the term `static` to `built_in`? Static makes me think that the instruments have been changed are stored in a static somewhere.

File: apollo-router/src/plugins/telemetry/mod.rs [hunk 2]
```diff
@@ -275,6 +281,57 @@ impl Plugin for Telemetry {
             ::tracing::warn!("telemetry.instrumentation.spans.mode is currently set to 'deprecated', either explicitly or via defaulting. Set telemetry.instrumentation.spans.mode explicitly in your router.yaml to 'spec_compliant' for log and span attributes that follow OpenTelemetry semantic conventions. This option will be defaulted to 'spec_compliant' in a future release and eventually removed altogether");
         }
 
+        let graphql_custom_instruments = if cfg!(test) {
```
Comment (BrynCooke) [inline apollo-router/src/plugins/telemetry/mod.rs:284]: Can we factor out this section of code and the related similar one in activate to a separate `fn` and remove the `cfg!(test)` logic? It doesn't matter if we instruments before activate as no-ops.

**Review by BrynCooke — CHANGES_REQUESTED** [review]

File: apollo-router/src/plugins/telemetry/config_new/instruments.rs (hunk 1 above)
Comment (bnjjj) [reply apollo-router/src/plugins/telemetry/config_new/instruments.rs]: Fixed https://github.com/apollographql/router/pull/5629/commits/c4a8b14787030df5190bcecc59b4d53fdeb99dfb

File: apollo-router/src/plugins/telemetry/mod.rs (hunk 2 above)
Comment (bnjjj) [reply apollo-router/src/plugins/telemetry/mod.rs]: Fixed https://github.com/apollographql/router/pull/5629/commits/c4a8b14787030df5190bcecc59b4d53fdeb99dfb

**Review by BrynCooke — APPROVED** [review]

**Pull request merged**